doc/ampi/manual.rst

```diff
@@ -519,53 +519,227 @@ provided that it takes place *after* ``MPI_Init``.
 Global Variable Privatization
 -----------------------------
 
-For the before-mentioned benefits to be effective, one needs to map
-multiple user-level threads onto each processor. Traditional MPI
-programs assume that the entire processor is allocated to themselves,
-and that only one thread of control exists within the process’s address
-space. So, they may safely use global and static variables in the
-program. However, global and static variables are problematic for
+For the aforementioned benefits to be effective, AMPI needs to map
+multiple user-level threads onto each OS process. Traditional MPI
```

Comment:
This makes it sound like only user-level threads are the issue, but it might also be kernel-level threads right?

Maybe:
```suggestion
AMPI maps multiple threads onto each OS process. Traditional MPI
```

Reply:
User-level threads alone are enough to cause privatization issues. The concurrency introduced by kernel threads could expose additional issues, but it might also obscure them nondeterministically. Maybe this section could say something like "AMPI maps multiple ranks onto each OS process in the form of user-level threads, which can execute cooperatively, concurrently, or some combination of both depending on how the Charm++ runtime system was built." Is that too much detail?

Reply:
> User-level threads alone are enough to cause privatization issues. The concurrency introduced by kernel threads could expose additional issues, but it might also obscure them nondeterministically. Maybe this section could say something like "AMPI maps multiple ranks onto each OS process in the form of user-level threads, which can execute cooperatively, concurrently, or some combination of both depending on how the Charm++ runtime system was built." Is that too much detail?

I think that might be a bit too verbose. Maybe:

"In AMPI, ranks are implemented as user-level threads that run on top of OS processes or OS threads, depending on the how the  Charm++ runtime was built."